src/intersections/rays.jl

```diff
@@ -149,58 +152,58 @@ function intersection(f, ray::Ray{3,T}, tri::Triangle{3,T}) where {T}
   det = e₁ ⋅ p
 
   # keep det > 0, modify T accordingly
-  if det > atol(T)
+  if det > atol(det)
     τ = o - vs[1]
   else
     τ = vs[1] - o
     det = -det
   end
 
-  if det < atol(T)
+  if det < atol(det)
     # This ray is parallel to the plane of the triangle.
     return @IT NotIntersecting nothing f
   end
 
   # calculate u parameter and test bounds
   u = τ ⋅ p
-  if u < -atol(T) || u > det
+  if u < -atol(u) || u > det
     return @IT NotIntersecting nothing f
   end
 
   q = τ × e₁
 
   # calculate v parameter and test bounds
   v = d ⋅ q
-  if v < -atol(T) || u + v > det
+  if v < -atol(v) || u + v > det
     return @IT NotIntersecting nothing f
   end
 
-  λ = (e₂ ⋅ q) * (one(T) / det)
+  λ = (e₂ ⋅ q) / det
```

Comment:
please make sure that this change is numerically equivalent, i.e., the order of operations with floating points doesn't change the result. we need to reduce the changes as much as possible in the PR to avoid introducing subtle bugs.